### DISCUSSION TITLE
Solid efforts overview

### DISCUSSION THREAD

**Discussion by elf-pavlik** [solid/specification#632] [General]
> [!WARNING]
> This was drafted before Solid-ODI and LWS WG were established.

This overview is by no means exhaustive. It is a starting point for the conversation and clarifying synergies between different efforts. I made this diagram, which is still a work in progress, I use following color code for the labels.

🔵 UX (all inclusive audience)
🟢 DX (broad dev audience)
🟠 DX (narrow dev audience)
🔴 feedback

use a button with double-sided arrow for a better preview

```mermaid
flowchart RL
  subgraph CG
    TR[Specifications]
    UCR[Use Cases & Requirements]
    QA[Test Suites]
    NOTE[Primers]
  end

  subgraph Lab?
    LIB[Shared Libraries]
    SERVICE[Core Services]
  end


  UCR ---|guide| TR
  NOTE ---|explain| TR
  QA ---|ensure| TR
  
  APP[Applications]
  USER[Users]


  subgraph Practicioners
    SERVICE ---|use| USER
    APP ---|use| USER
  end
  
  USER -..-|provide| UCR
  LIB ---|use| SERVICE
  LIB ----|use| APP

  QA ---|verify| LIB
  QA ---|verify| SERVICE

  NOTE ----|inform| SERVICE
  NOTE ----|inform| APP
  NOTE ----|inform| LIB
  
  SERVICE ---|implement| TR
  TR ---|implement| LIB
  
  linkStyle 5 color:pink;
  linkStyle 6,8,9,10,12,13,14 color:orange;
  linkStyle 3,4 color:lightblue;
  linkStyle 7,11 color:lightgreen;
```
 
So far my main takeaways are:

* We need to have clarity on who is the audience for each of the deliverables
  * users are practically everyone
  * App developers will be the biggest dev group, and they shouldn't be expected to dive into nitty-gritty nuances; they can rely on simple primers and shared libraries
  * Core services and shared libraries dev groups will be relatively small; they need to understand the specs and be able to use conformance tests.
 * We need to work with those audiences to ensure we are addressing precise requirements from real-world use cases. Each having aspects related to
   * usability / accessibility - e.g. regular users will not be editing turtle files and often will use mobile devices
   * privacy
   * security
   * sovernity / portability
   * etc.

**Comment by jeff-zucker** [comment]
This is very helpful.  Minor gripe - my color-blindness makes the green and blue labels almost invisible against the dark gray.

This is very similar to the way I describe things in the [Practitioners README](https://github.com/solid-contrib/practitioners).  I envision two other panels to add - one for new developers - this is a very different group with different needs from devs actively working on real world apps and needs tutorials and other resources not shown on the diagram.  Hopefully someone will step up and lead the Onboarding task force and start regular meetups for this population.  Another panel needed to complete a view of the entire ecosystem is one for "everyone else" - which needs at least two pathways - outreach to draw them in as either developers or adopters.  Again that audience needs a different set of resources from the ones shown.

**Reply by elf-pavlik** [reply]
> the green and blue labels almost invisible against the dark gray 

I adjusted them to lighter versions.

**Reply by jeff-zucker** [reply]
Much better, thanks.

**Comment by jeff-zucker** [comment]
I am thinking about the way feedback gets into the system.  Do users actually come to the CG and present use cases?  Or is that more likely to be developers?  I don't think the homeless people who use the PASS project will be showing up at the CG.  Instead the homeless people will report problems to PASS project devs (they already have feedback mechanisms in place) and the devs from that project will report it either to the CG via Practitioners or directly to the CG.

**Reply by elf-pavlik** [reply]
I see feedback bubbling up the UX -> DX chain. I changed the line on the diagram to a dotted style.

**Comment by CxRes** [comment]
Just a suugestion. I would convert this to a sandwich with service providers and library builders (ie lab?) as the cheese and tomato partially covering between the broad CG and User bread. Something like a primer is at the top of the CG bread but specs at the very bottom.

**Reply by elf-pavlik** [reply]
The layout is automatic, but one still has some degree of control. You can use the button to copy the source, paste it into the mermaid code block, and try to see if you can get what you are looking for. The order of things affects how they are automatically organized.
